### PR TITLE
sync: rollbacks db on exception

### PR DESCRIPTION
**Pull request by jrcastro2**

### CODE REVIEW

File: invenio_cern_sync/users/sync.py [hunk 1]
```diff
@@ -188,14 +188,18 @@ def _insert_missing(invenio_users, log_uuid, log_name):
                     f"Skipping user with username starting with `_`: {invenio_user}"
                 )
                 continue
-            _id = create_user(invenio_user)
+
+            with db.session.begin_nested():
+                _id = create_user(invenio_user)
+                db.session.flush()
```
Comment (ntarocco) [inline invenio_cern_sync/users/sync.py:194]: This should not be needed? the `begin_nested` will commit the transaction `on_exit`
```suggestion

```

File: invenio_cern_sync/users/sync.py [hunk 2]
```diff
@@ -188,14 +188,18 @@ def _insert_missing(invenio_users, log_uuid, log_name):
...
+                inserted.add(_id)
+
+            db.session.commit()
```
Comment (ntarocco) [inline invenio_cern_sync/users/sync.py:197]: each `begin_nested` should already commit each time
```suggestion

```

File: invenio_cern_sync/users/sync.py (hunk 1 above)
Comment (jrcastro2) [reply invenio_cern_sync/users/sync.py:194]: Right, it should flush before entering and on exit, so we should be safe, thanks 👍

File: invenio_cern_sync/users/sync.py (hunk 2 above)
Comment (jrcastro2) [reply invenio_cern_sync/users/sync.py:197]: Hmm, If I understand correctly the begin_nested does "commit" the SAVEPOINT meaning that it flushes those changes and does release the SAVEPOINT but doesn´t do the actual commit to the DB until we call db.session.commit()

> When the context manager yielded by [Session.begin_nested()](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.begin_nested) completes, it “commits” the savepoint, which includes the usual behavior of flushing all pending state.

> When the .commit() method on this object is called, “RELEASE SAVEPOINT” is emitted to the database, and if instead the .rollback() method is called, “ROLLBACK TO SAVEPOINT” is emitted. The enclosing database transaction remains in progress.

Comment (ntarocco) [reply invenio_cern_sync/users/sync.py:196]: You are right, it is only "closing" the transaction/savepoint.
However, we are now basically isolating the transaction, and potentially rollback. If one fails, the other will still work and succeed with a single `db.session.commit()` at the end.
I think we should find a tradeoff: what about having a commit after e.g. 1000 users?

**Review by ntarocco — APPROVED** [review]

**Pull request merged**